### ISSUE TITLE
No on end when data stream is not used

### ISSUE THREAD

**Issue by wzrdtales** [mscdex/node-mariasql#108]
Currently the on end event does not  get called if one executes a statement like Start transaction

Fox example given this example call:

```javascript
    return new Promise( function( resolve, reject ) {

      conn.query( 'START TRANSACTION;' )
      .on( 'result', function ( res ) {

        res.on( 'error', function ( err ) {

          reject( err );
        } )
        .on( 'end', function ( info ) {

          conn._is_in_transaction = true;
          resolve( res.info );
        } );
      } );
    } );
```

This is problematic, because this makes it impossible to catch on the on error event.

**Comment by wzrdtales** [comment]
@mscdex Update, just figered out that this occurs if one leave out the data event. In this case I do not want to access this event, this would at least need a mention within the documentation if not considered as bug.

**Comment by mscdex** [comment]
Should be fixed in 5fbb8dcd43.
